web/frontend/src/components/slam/2d-renderer.vue

```diff
@@ -143,7 +143,11 @@ const updatePose = (newPose: commonApi.Pose) => {
   const x = newPose.getX();
   const y = newPose.getY();
   const z = newPose.getZ();
+
   baseMarker.position.set(x, y, z);
+
+  const theta = THREE.MathUtils.degToRad(newPose.getTheta());
+  baseMarker.geometry.rotateZ(theta);
```

Comment:
This is pretty much it. Three.js works in radians and Viam in degrees so I had to convert. Sprites always face the camera directly with rotations ignored so their geometry, rather than Object3d, must be rotated.